apps/editing-toolkit/editing-toolkit-plugin/tutorials/class-wpcom-tutorials.php

```diff
@@ -0,0 +1,227 @@
+<?php
+/**
+ * Tutorials (We would have called them Guided Tours but it's been done).
+ *
+ * Each Tutorial contains a flat list of Tasks. Each Task may contain several Steps in
+ * a `WpcomTourKit` component on the frontend, but we are only concerned with the top
+ * two levels of Tutorial and its contained Tasks, tracking Task completion status
+ * on the backend. This API will be used to generate
+ *
+ * @package A8C\FSE
+ */
+
+/**
+ * WPCom_Tutorials class
+ */
+class WPCom_Tutorials {
+
+	/**
+	 * Storage for our singleton instance.
+	 *
+	 * @var null|WPCom_Tutorials
+	 */
+	private static $instance = null;
+
+	/**
+	 * Internal storage for registered Tutorials.
+	 *
+	 * @var array
+	 */
+	private $registry = array();
+
+	/**
+	 * Gets aour singleton instance, initiating it the first time.
+	 *
+	 * @return WPCom_Tutorials instance.
+	 */
+	public static function get_instance() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Registers a Tutorial.
+	 *
+	 * @param string $id      A unique ID for the Tutorial.
+	 * @param array  $options `title` and `tasks` required.
+	 *
+	 * @return bool           True if successfully registered, false if it failed.
+	 */
+	public function register( $id, $options ) {
+		// @todo verify tasks strutcture
+		if ( ! $this->validate_options( $options ) ) {
+			return false;
+		}
+		$options['id']         = $id;
+		$this->registry[ $id ] = $options;
+		return true;
+	}
+
+	/**
+	 * Unregisters a registered Tutorial
+	 *
+	 * @param string $id Tutorial ID.
+	 * @return bool      True if successfully unregistered, false if not found.
+	 */
+	public function unregister( $id ) {
+		if ( ! isset( $this->registry[ $id ] ) ) {
+			return false;
+		}
+		unset( $this->registry[ $id ] );
+		return true;
+	}
+
+	/**
+	 * Validates the options passed in `WPCom_Tutorials::register`.
+	 *
+	 * @param array $options Associative array passed with registration.
+	 * @return bool          Validation success.
+	 */
+	private function validate_options( $options ) {
+		if ( ! isset( $options['title'] ) || ! is_string( $options['title'] ) ) {
+			return false;
+		}
+		if ( ! isset( $options['tasks'] ) || ! is_array( $options['tasks'] ) ) {
+			return false;
+		}
+		foreach ( $options['tasks'] as $task ) {
+			if ( ! $this->validate_task( $task ) ) {
+				return false;
+			}
+		}
+		return true;
+	}
+
+	/**
+	 * Validates a single task for expected structure.
+	 *
+	 * @param array $task A single task (id and title).
+	 * @return bool       Validation success.
+	 */
+	private function validate_task( $task ) {
+		return isset( $task['id'] ) && isset( $task['title'] );
+	}
+
+	/**
+	 * Retrieves a Tutorial from the registry.
+	 *
+	 * @param string $id  The ID of the Tutorial to fetch.
+	 * @return null|array Tutorial associative array if found, null if not.
+	 */
+	public function get( $id ) {
```

Comment:
`get_tutorial` might be a little more semantic.

Reply:
It would be. My own thinking was that it's implied by `Tutorials::get` but a few extra bytes will cost me nothing and the extra clarity is a kindness to future folks needing to read this.